//  File: tao_rpc.proto
//  Author: Tom Roeder <tmroeder@google.com>
//
//  Description: Protocol buffers for TaoRPC.
//
//  Copyright (c) 2013, Google Inc.  All rights reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package tao;

// Copied from: util/protorpc/protorpc.proto

// Protobuf RPC request header
message ProtoRPCRequestHeader {
  // The service method.
  required string op = 1;

  // The sequence number.
  required uint64 seq = 2;
}

// Protobuf RPC response header.
message ProtoRPCResponseHeader {
  // The service method (matches request op).
  required string op = 1;

  // The sequence number (matches request seq).
  required uint64 seq = 2;

  // The optional error string.
  optional string error = 3;
}

// Copied from: tao/tao_rpc.proto

message TaoRPCRequest {
  optional bytes data = 1;
  optional int32 size = 2;
  optional string policy = 3;
  optional int64 time = 4;
  optional int64 expiration = 5;
  optional bytes issuer = 6;
  // optional string label = 7;
  // optional int64 counter = 8;
}

message TaoRPCResponse {
  optional bytes data = 1;
  optional string policy = 2;
  // optional int64 counter = 3;
}
